mailbox/cassandra/src/test/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdDAOTest.java

```diff
@@ -0,0 +1,174 @@
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.cassandra.mail;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Optional;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.mailbox.MessageUid;
+import org.apache.james.mailbox.cassandra.CassandraId;
+import org.apache.james.mailbox.cassandra.CassandraMessageId;
+import org.apache.james.mailbox.cassandra.modules.CassandraMessageModule;
+import org.apache.james.mailbox.model.MessageRange;
+import org.apache.james.mailbox.store.mail.MessageMapper.FetchType;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.datastax.driver.core.utils.UUIDs;
+
+public class CassandraMessageIdDAOTest {
+
+    private CassandraCluster cassandra;
+
+    private CassandraMessageIdDAO testee;
+
+    @Before
+    public void setUp() {
+        cassandra = CassandraCluster.create(new CassandraMessageModule());
+        cassandra.ensureAllTables();
+
+        testee = new CassandraMessageIdDAO(cassandra.getConf());
+    }
+
+    @After
+    public void tearDown() {
+        cassandra.clearAllTables();
+    }
+
+    @Test
+    public void deleteShouldNotThrowWhenRowDoesntExist() {
+        testee.delete(CassandraId.timeBased(), MessageUid.of(1))
+            .join();
+    }
+
+    @Test
+    public void deleteShouldDeleteWhenRowExists() {
+        CassandraId mailboxId = CassandraId.timeBased();
+        MessageUid messageUid = MessageUid.of(1);
+        CassandraMessageId messageId = CassandraMessageId.of(UUIDs.timeBased());
+        testee.insert(mailboxId, messageUid, messageId).join();
+        Optional<CassandraMessageId> insertedMessages = testee.retrieve(mailboxId, messageUid).join();
+        assertThat(insertedMessages.get()).isEqualTo(messageId);
+
+        testee.delete(mailboxId, messageUid).join();
+
+        Optional<CassandraMessageId> message = testee.retrieve(mailboxId, messageUid).join();
+        assertThat(message.isPresent()).isFalse();
```

Comment:
I've tried, but failed....
I had upgraded assertj to 3 (java 8), and implement my test with this.
But then mappers tests (which are on a java 6 project -> assertj 1) are failing due to a different API on `assertThat(Map)`.
I didn't go further more, each implementations have to provide his assertThat API... to much complexity for two assertions.